tests/arch/arm/arm_irq_vector_table/src/arm_irq_vector_table.c

```diff
@@ -152,24 +166,34 @@ void test_arm_irq_vector_table(void)
 
 typedef void (*vth)(void); /* Vector Table Handler */
 
-#if defined(CONFIG_SOC_SERIES_NRF52X) || defined(CONFIG_SOC_SERIES_NRF91X)
-/* nRF52X- and nRF91X-based platforms employ a Hardware RTC peripheral
+#if defined(CONFIG_SOC_FAMILY_NRF)
+/* nRF5X- and nRF91X-based platforms employ a Hardware RTC peripheral
  * to implement the Kernel system timer, instead of the ARM Cortex-M
  * SysTick. Therefore, a pointer to the timer ISR needs to be added in
  * the custom vector table to handle the timer "tick" interrupts.
  *
  * The same applies to the CLOCK Control peripheral, which may trigger
  * IRQs that would need to be serviced.
+ *
+ * Note: qemu_cortex_m0 uses TIMER0 to implement system timer.
  */
 void rtc1_nrf_isr(void);
 void nrf_power_clock_isr(void);
-#if defined(CONFIG_SOC_SERIES_NRF52X)
+#if defined(CONFIG_SOC_SERIES_NRF51X) || defined(CONFIG_SOC_SERIES_NRF52X)
+#if defined(CONFIG_BOARD_QEMU_CORTEX_M0)
+void timer0_nrf_isr(void);
+vth __irq_vector_table _irq_vector_table[RTC1_IRQn + 1] = {
+	nrf_power_clock_isr,
+	0, 0, 0, 0, 0, 0, 0, timer0_nrf_isr, isr0, isr1, isr2, 0, 0, 0, 0, 0,
```

Comment:
These trailing `0`s don't seem to be useful.

Reply:
Thanks, will fix